granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -70,4 +72,29 @@ def find_v2_hierarchy() -> Optional[str]:
         return None
     if len(cgroup2_mounts) > 1:
         raise Exception("More than one cgroup2 mount found!")
-    return ns.resolve_host_root_links(cgroup2_mounts[0].mount_point)
+    path = cgroup2_mounts[0].mount_point
+    if resolve_host_root_links:
+        path = ns.resolve_host_root_links(path)
+    return path
+
+
+def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> Optional[str]:
+    """
+    Returns the folder that the requested controller is mounted to, or None if no such controller mount was found
+    If no v1 mount was found for the requested controller and there is a v2 mount (either unified or hybrid),
+    the v2 mountpoint is returned (as all controllers share the same hierarchy in v2)
+    """
+    v1_paths = find_v1_hierarchies(resolve_host_root_links)
+    v2_path = find_v2_hierarchy(resolve_host_root_links)
+    if v1_paths and controller in v1_paths:
+        # Either v1 or hybrid with the requested controller bound to a v1 hierarchy
+        return v1_paths[controller]
+    if v2_path:
+        # v2 (unified) - all controllers are in a single unified hierarchy
+        return v2_path
+    # No cgroup mount of the requested controller
+    return None
+
+
+def is_valid_controller(controller: str) -> bool:
+    return controller in SUBSYSTEMS
```

Comment:
"valid" is a bit misleading. Perhaps "known"? This also does not check if the controller is actually in use on the system, so it's unclear what constitutes an "invalid" controller. A list of enabled subsystems is in `/proc/cgroups` if we actually care about that.

Reply:
Sure I can go for `known`. This isn't supposed to check if the controller is in use (mounted), but rather if the given string is a recognized cgroup controller